src/components/admin/Billing/PricingTierDetails.tsx

```diff
@@ -7,22 +7,33 @@ import { useTenantStore } from 'stores/Tenant/Store';
 
 function PricingTierDetails() {
     const selectedTenant = useTenantStore((state) => state.selectedTenant);
-    const externalPaymentMethod = useTenantUsesExternalPayment(selectedTenant);
+    const [externalPaymentMethod, marketPlaceProvider] =
+        useTenantUsesExternalPayment(selectedTenant);
 
     const billingStoreHydrated = useBillingStore((state) => state.hydrated);
     const paymentMethodExists = useBillingStore(
         (state) => state.paymentMethodExists
     );
 
-    const messageId = useMemo(
-        () =>
-            externalPaymentMethod
-                ? 'admin.billing.message.external'
-                : paymentMethodExists
-                ? 'admin.billing.message.paidTier'
-                : 'admin.billing.message.freeTier',
-        [externalPaymentMethod, paymentMethodExists]
-    );
+    const messageId = useMemo(() => {
+        if (externalPaymentMethod) {
+            if (marketPlaceProvider === 'gcp') {
+                return 'admin.billing.message.external.gcp';
+            }
+
+            if (marketPlaceProvider === 'aws') {
+                return 'admin.billing.message.external.aws';
+            }
+
+            return 'admin.billing.message.external';
```

Comment:
This could be simplified to one of the following to increase scalability:

```ts
// If the ID of the generic content had an identifying suffix.
return `admin.billing.message.external.${marketPlaceProvider ?? 'generic }`;
```

OR

```ts
return marketPlaceProvider ? `admin.billing.message.external.${marketPlaceProvider}` : 'admin.billing.message.external';
```

Reply:
unless it is deeply important we should try not to dynamically build translate strings.